app/components/ui/BookmarkModal/BookmarkModal.tsx

```diff
@@ -0,0 +1,216 @@
+              >
+                <i className={`material-icons ${styles.clearBookmarkText}`}>
+                  cancel
+                </i>
+              </button>
+            </div>
+            <p className={styles.bookmarkUrl}>{`${siteUrl}/${
+              hit.type === "service" ? `services` : `organizations`
+            }/${hit.id}`}</p>
+          </div>
+          <p className={styles.label}>LOCATION</p>
+          <div
+            className={styles.selectFolderContainer}
+            onBlur={(evt) => {
+              handleDropwodnBlur(evt);
+            }}
+            // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
+            tabIndex={0}
+            ref={foldersContainerRef}
+          >
+            {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions */}
+            <p
+              className={styles.selectFolder}
+              onClick={() => setExpandFolders(!expandFolders)}
+            >
+              <span className={selectedFolder?.name ? "" : styles.placeholder}>
+                {selectedFolder?.name ?? "Select Folder"}
+              </span>
+              <i className="material-icons">arrow_drop_down</i>
+            </p>
+            {expandFolders && (
+              <ul className={styles.bookmarkFoldersList}>
+                {folderOptions.map((folder) => {
+                  return (
+                    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+                    <li
+                      key={folder.id}
+                      className={styles.folderListItem}
+                      onClick={() => {
+                        setSelectedFolder(folder);
+                        setExpandFolders(false);
+                      }}
+                    >
+                      <i className={`material-icons ${styles.folderIcon}`}>
+                        folder
+                      </i>
+                      {folder.name}
+                    </li>
+                  );
+                })}
+                <li
+                  className={`${styles.folderListItem} ${styles.addFolderItem}`}
+                >
+                  <label
+                    className={styles.addFolderLabel}
+                    htmlFor="createNewFolder"
+                  >
+                    +
+                  </label>
+                  <input
+                    id="createNewFolder"
+                    className={styles.addFolderInput}
+                    type="text"
+                    placeholder="Create new folder"
+                    value={newFolder}
+                    onChange={(evt) => createNewFolder(evt.target.value)}
+                    onFocus={(evt) => {
+                      createNewFolder(evt.target.value);
+                    }}
+                    onKeyDown={(evt) => {
+                      if (evt.key === "Enter") {
+                        setExpandFolders(false);
+                      }
+                    }}
```

Comment:
I feel that the way these events are hooked up would be a bit confusing to me as a user. Like I mentioned earlier, there are two separate actions to take here: setting the name of the new folder, and selecting the new folder as the location for the bookmark. The way you have it here, both typing into the textbox and moving the focus onto the textbox (either by clicking or tabbing) will simultaneously take both actions, while hitting Enter will merely hide the drop-down menu.

I wonder if instead we did the following:

- onChange only changes the name of the new folder, without selecting it
- onFocus does nothing
- onKeyDown (evt.key === "Enter") actually sets the new folder as the selected folder and hides the folders

I think this would make it more similar to the other, non-new folder options, where the action of "selecting" a folder (via clicking) causes the selected folder to change and hides the menu.

I am also wondering if we should attach a click event handler on the `+` button, to have that be able to respond to events. For the normal, non-new folder options, you have the event handler on the entire `<li>` element.

The more I type, the more I'm second-guessing myself, and at this point, I'm not even sure if you should bother making any of the changes I'm suggesting, since it's probably best for us to clarify with a designer.